### PR TITLE
Ensure pod inspect is locked and validity-checked

### PR DESCRIPTION
**Pull request by mheon**
Also, don't return the internal podState struct - instead return a public inspect struct.

### CODE REVIEW

**Comment by baude** [comment]
LGTM

File: libpod/pod.go [hunk 1]
```diff
@@ -50,17 +50,22 @@ type PodConfig struct {
 // podState represents a pod's state
 type podState struct {
 	// CgroupPath is the path to the pod's CGroup
-	CgroupPath string
+	CgroupPath string `json:"cgroupPath"`
```
Comment (TomSweeneyRedHat) [inline libpod/pod.go]: podState is an odd name for a strut that only has a path in it....

Comment (mheon) [reply libpod/pod.go]: It will eventually have more - I think @haircommander is adding a container ID for the pause container in his PR.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**